### PR TITLE
Set error status if http has exception and ok status.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
InvokeAsync middleware doesn't receive the correct status code during an error, setting a transaction with an error as "OK", this PR sets the default status code based on the received Exception and if the status code is Ok.

Before
image

After
image

### CODE REVIEW

**Comment by Tyrrrz** [comment]
Does this happen because the status code is set after the middleware?

**Review by Tyrrrz — APPROVED** [review]

**Comment by lucas-zimerman** [comment]
> Does this happen because the status code is set after the middleware?

yup

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 ## Unreleased
 
-## Fixes
+### Fixes
 
+- Set error status to transaction if http has exception and ok status ([#1143](https://github.com/getsentry/sentry-dotnet/pull/1143))
```
Comment (bruno-garcia) [inline CHANGELOG.md:7]:
```suggestion
- End transaction with error if exception is thrown in pipeline ([#1143](https://github.com/getsentry/sentry-dotnet/pull/1143))
```

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs [hunk 2]
```diff
@@ -157,6 +157,10 @@ public async Task InvokeAsync(HttpContext context)
                     {
                         transaction.Finish(status);
                     }
+                    else if (status == SpanStatus.Ok)
+                    {
+                        transaction.Finish(exception);
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/SentryTracingMiddleware.cs:162]:
```suggestion
                    // Status code not yet changed to 500 but an exception does exist
                    // so lets avoid passing the misleading 200 down and close only with
                    // the exception instance that will be inferred as errored
                    else if (status == SpanStatus.Ok && exception is not null)
                    {
                        transaction.Finish(exception);
```

**Review by bruno-garcia — CHANGES_REQUESTED** [review]
I think we need to check if we have an instance because passing `null` as exception could behave as `unknown` status or as `ok` or that can change in the future.

There are no tests for this so sounds like we can change this and nothing validates the change.

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs (hunk 2 above)
Comment (lucas-zimerman) [reply src/Sentry.AspNetCore/SentryTracingMiddleware.cs:162]: Doesn't the previous lines guarantee that you don't have a null exception?
```csharp
                    if (exception is null)
                    {
                        transaction.Finish(status);
                    }
                    else if (status == SpanStatus.Ok)
                    {
                        transaction.Finish(exception);
                    }
                    else
                    {
                        transaction.Finish(exception, status);
                    }
```

Comment (lucas-zimerman) [reply src/Sentry.AspNetCore/SentryTracingMiddleware.cs:162]: Slightly edited the test Transaction_binds_exception_thrown for checking if the Span was set as Internal Error.
Without the changes in this PR, the test was returning a span with the Status Ok.

Comment (Tyrrrz) [reply src/Sentry.AspNetCore/SentryTracingMiddleware.cs:162]: Yeah it already checks for null

Comment (bruno-garcia) [reply src/Sentry.AspNetCore/SentryTracingMiddleware.cs:162]: OK, please add the comments

File: test/Sentry.AspNetCore.Tests/SentryTracingMiddlewareTests.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿#if !NETCOREAPP2_1
+#if !NETCOREAPP2_1
```
Comment (bruno-garcia) [inline test/Sentry.AspNetCore.Tests/SentryTracingMiddlewareTests.cs:1]: BOM?

**Review by bruno-garcia — APPROVED** [review]

File: test/Sentry.AspNetCore.Tests/SentryTracingMiddlewareTests.cs (hunk 3 above)
Comment (lucas-zimerman) [reply test/Sentry.AspNetCore.Tests/SentryTracingMiddlewareTests.cs:1]: not sure what has changes, at least, vs code doesn't spot it https://github1s.com/getsentry/sentry-dotnet/pull/1143

**Pull request merged**